extensions/core/interfaces/icons/interface.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <div class="interface-icons">
+    <v-input v-model="searchText" placeholder="Search an icon" :icon-right="value"/>
+    <div class="icons-view" v-show="searchText.length === 0">
+      <details 
+      v-for="(icongroup,groupname) in icons"
+      :key="(icongroup,groupname)"
+      open>
+        <summary>
+          {{groupname}}
+        </summary>
+        <div>
+          <button
+            v-for="icon in icongroup"
+            :key="icon"
+            v-tooltip="$helpers.formatTitle(icon)"
+            :class="{ active: value === icon}"
+            @click="$emit('input', icon)">
+            <i class="material-icons">{{ icon }}</i>
+          </button>
+        </div>
+      </details>
+    </div>
+    <div class="search-view" v-if="searchText.length > 0">
+      <button
+        v-for="icon in filteredArray"
+        v-tooltip="$helpers.formatTitle(icon)"
+        :key="icon"
+        :class="{ active: value === icon}"
+        @click="$emit('input', icon)">
+        <i class="material-icons">{{ icon }}</i>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import mixin from "../../../mixins/interface";
+import icons from "./icons.json";
+
+export default {
+  mixins: [mixin],
+  data() {
+    return {
+      searchText: ""
+    };
+  },
+  computed: {
+    icons() {
+      return icons;
+    },
+    iconsArray() {
+      return this.$lodash.flatten(Object.values(this.icons));
+    },
+    filteredArray() {
+      return this.iconsArray.filter(icon =>
+        icon.includes(this.searchText.toLowerCase())
+      );
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+.interface-icons {
+  display: flex;
+  flex-direction: column;
+  overflow-y: scroll;
+  height: 320px;
+  width: 100%;
+  max-width: var(--normal);
+  border: var(--input-border-width) solid var(--lighter-gray);
+  border-radius: var(--border-radius);
+  background-color: white;
+  padding: 0.3em;
+
+  .v-input {
+    color: var(--gray);
+    height: var(--input-height);
+    transition: var(--fast) var(--transition);
+    transition-property: color, border-color, padding;
+    position: sticky;
+    top: 0;
+    z-index: +1;
+
+    &:focus {
+      color: var(--darker-gray);
+      border-color: var(--accent);
+      outline: 0;
+    }
+  }
+  .icons-view {
+    details {
+      text-transform: uppercase;
+      summary {
+        margin: 5px 2px;
+      }
+      div {
+        button {
+          padding: 0.4em;
```

Comment:
We use intervals of 10px everywhere in the system for margin / padding. Lets switch these to px